### PR TITLE
22781 logistic jax frontend

### LINKED ISSUE #22781: logistic

### PR DESCRIPTION
**Pull request by rsaeta**
# PR Description
Adds logistic function to jax frontend for `jax.random.logistic`.

## Related Issue

Close #22781

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you follow the steps we provided?

Implementation is based on performance testing of a number of different implementations. 
Assuming `uniform_x = ivy.random_uniform(seed=seed, shape=shape, dtype=dtype)`, to generate 1M numbers:

`ivy.log(uniform_x) - ivy.log(1 - uniform_x)` takes 0.1869s
`ivy.log(uniform_x / (1 - uniform_x))` takes 0.0193s
`ivy.log(ivy.divide(x, ivy.subtract(1.0, x)))` takes 0.0094s

### CODE REVIEW

**Review by AnnaTz — APPROVED** [review]
Looks good! Thanks!
